### ISSUE TITLE
[Mod request] Realistic hang glider

### ISSUE THREAD

**Issue by TheEt1234** [pandorabox-io/in-game#380]
https://content.minetest.net/packages/CBugDCoder/glider/

Basically our current glider but like way way way more fun (and way more risky)

Requires tnt mod for rocket recipe (where you can go 500 nodes/s and crash into a mountain [and die])

**Comment by TheEt1234** [comment]
https://github.com/pandorabox-io/in-game/assets/74461678/02667431-ea43-44e6-b8b0-c2a518a5e6f6

**Comment by TheEt1234** [comment]
https://github.com/pandorabox-io/in-game/assets/74461678/6e29ba4d-b1b9-4659-a75e-f91a782b3b53

**Comment by SwissalpS** [comment]
I doubt the server can handle 500 n/s :p
Also, that would probably trigger a watchdog (which shouldn't be a problem to solve I think)

gunpowder, needed for rockets, shouldn't be a problem.

I don't see why the current glider would have to go, both could be installed at the same time, just call this one hangglider.

**Comment by TheEt1234** [comment]
> I doubt the server can handle 500 n/s :p
> Also, that would probably trigger a watchdog (which shouldn't be a problem to solve I think)

right.... maybe disabling rockets would be a sane idea....  or making them slower (maybe a speed cap?)

> gunpowder, needed for rockets, shouldn't be a problem.


> I don't see why the current glider would have to go, both could be installed at the same time, just call this one hangglider.


It doesn't have to go, yes

**Comment by SwissalpS** [comment]
I just realized, the one in use now is already called hangglider, so there wouldn't even be a conflict there.

**Comment by SwissalpS** [comment]
still, it would create a bunch of confusion about the names. Similar to 'inspection tool' vs 'node inspection tool' only that more players will be using these (and have access to)
I have forked the glider and am seeing if it can be coloured and also the other advanced functions of [hangglider] can be implemented: flak, can_fly, tool wear.

**Comment by SwissalpS** [comment]
WIP but ready for beta testers:
https://github.com/SwissalpS/minetest_glider

It should be more stable than the one on CDB. The use of rockets can be disabled per server and if enabled, there is a delay setting so players can't use a barrage of rockets.
Maybe there will still need to be a speed limiter. On my machine I've achieved a max speed of 30 (before limiting rockets).

I've also included features from [hangglider] like colours and FLAK.

(I'm still not sure this is a good match for pandorabox)

**Comment by TheEt1234** [comment]
Why it wouldn't be a good match for pandorabox?

**Comment by SwissalpS** [comment]
> Why it wouldn't be a good match for pandorabox?

exactly, why would it be a good match?

Main reason why not, is there is already the hangglider, xenos, horses, digtrons that can be used to travel around swiftly.
(Also trains, travelnets, compass, telemosaic, two types of elevators, jumpdrives)
There have been discussions in the past, of how to nerf some of them to improve the way cities are built, and also to reduce the amount of mapblocks the server needs to load.
With [glider] players can zoom through many blocks in very short amount of time. While [hangglider] is mostly used to protect against fall damage and slow down the amount of blocks passed, [glider] does pretty much the opposite. It's very much a mini-game in itself that invites to travel large distances and just have fun flying. A flightsim in multiplayer could be rather heavy on the server. (even without rockets).

More savy people will have to test these hypothosis' out and prove my guesswork wrong.

Don't get me wrong, I love flight simulations, partly why I was so motivated to improve the mod. I'm just  a bit concerned about the server load and a little also about the confusion of having two so similarly named items with quite different use cases.

If [glider] does get added, I would PR a way to not allow users to use both at the same time.

**Comment by TheEt1234** [comment]
> With [glider] players can zoom through many blocks in very short amount of time.

I see, *if* this is a big issue, maybe its speed could be nerfed

> exactly, why would it be a good match?
> Main reason why not, is there is already the hangglider, xenos, horses, digtrons that can be used to travel around swiftly.
(Also trains, travelnets, compass, telemosaic, two types of elevators, jumpdrives)

Yes but this glider is unique because it isn't teleporting and it's a beautiful way to explore new terrain or cities from the sky, and as you mentioned it can be used as a mini-game

> and a little also about the confusion of having two so similarly named items with quite different use cases.

i don't think this will be an issue

**Comment by SwissalpS** [comment]
> it's a beautiful way to explore new terrain or cities from the sky

that sounds like [hangglider], with [glider] my experience has been more of: crashing in weird places and trying to figure out how to get back for my bones :p

[glider] passes through cities so quickly it's hard to call that sight-seeing. Especially when pilot is more concerned about not crashing  :)

> maybe its speed could be nerfed

That's still something I haven't exactly added to the mod yet. To have lift there needs to
be speed, so just capping speed, isn't all. The lift formula needs to be adjusted depending on speed cap.
At least there already is a rocket-cooldown-delay.

Anyway it's not my decission luckily :D

**Comment by TheEt1234** [comment]
> [glider] passes through cities so quickly it's hard to call that sight-seeing. Especially when pilot is more concerned about not crashing :)

> crashing in weird places and trying to figure out how to get back for my bones :p

well, if you have fast priv or wear armor that makes you go fast or use rockets that make you go fast, then yeah the glider will be fast...

With speed there is risk and i think that's a good thing, but we will probably need more feedback on this

~~also silly idea: make a "rocket" that makes you slower~~ maybe the sneak button would be better for that

**Comment by SwissalpS** [comment]
> ...makes you slower maybe the sneak button would be better for that

just use [hangglider] to go slow. [glider] needs more speed to stay airborne.
You 'can' go slow with [glider] just by lifting nose too much. That looses speed quickly.

**Comment by SwissalpS** [comment]
> With speed there is risk and i think that's a good thing

Yes risk a good thing. It's not what I'm concerned about the speed. It's the server and network load that speed entails.

**Comment by TheEt1234** [comment]
right... what would be an acceptable speed

i think roughly little less than speed beacon+normal hangglider+lava boots would be good(?)

so a speed of 10 maybe? (beacon+lava boots+hangglider is 14.7, you can measure this with a client side mod)

**Comment by SwissalpS** [comment]
Bump: [deltaglider] has been improved and now maintained by mt-mods. https://github.com/mt-mods/deltaglider

It's now worth testing on testserver and deciding about adding or closing this issue.
